python/svix/api/common.py

```diff
@@ -22,11 +27,32 @@ def sanitize_field(v: t.Any) -> t.Any:
     return v
 
 
+def _serialize_single_param(val: t.Any) -> str:
+    if isinstance(val, datetime):
+        if val.tzinfo is None:
+            val.replace(tzinfo=timezone.utc)
+        return val.isoformat()
+    elif isinstance(val, bool):
+        return "true" if val else "false"
+    else:
+        return str(val)
+
+
+def serialize_params(d: t.Dict[str, t.Optional[t.Any]]) -> t.Dict[str, str]:
+    return {k: _serialize_single_param(v) for k, v in d.items() if v is not None}
+
+
 @dataclass
 class BaseOptions:
     def to_dict(self) -> t.Dict[str, t.Any]:
         return {k: sanitize_field(v) for k, v in asdict(self).items() if v is not None}
 
+    def _query_params(self) -> t.Dict[str, str]:
+        return {}
+
+    def _header_params(self) -> t.Dict[str, str]:
+        return {}
```

Comment:
```suggestion
    def _query_params(self) -> t.Optional[t.Dict[str, str]:
        return None

    def _header_params(self) -> t.Optional[t.Dict[str, str]:
        return None
```
It looks like None is accepted by httpx helpers? If so, we can just do that for the empty case.

Reply:
I think it makes sense to keep the signature consistent with the generated `_query_params`/`_header_params` method, its signature is `def _query_params(self) -> t.Dict[str, str]:`.
And I don't think it makes sense to have the generated function hint a `t.Optional[t.Dict[..]]` if it will always return a `t.Dict[..]`

Reply:
It doesn't always return a dict, if it's OOP and the base function returns None so it's very consistent. That's how people do function overloading in OOP when the default is None. I don't want to nitpick too much, so do whatever, just calling it out.